Pull Req: Added --params option to pass additional params to downstream tools

rwmtse opened the pull request:
User Story: blackberry/BB10-Webworks-Packager#157
CI Build: http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-params-tools/5/
Microsite Issue/Sample Usage: http://github.rim.net/Microsites/HTML5-WebWorks/issues/193

@issp

rwmtse commented:
@adrianlee Can you code review?

adrianlee commented on lib/signing-helper.js:
```diff
@@ -48,6 +53,50 @@ function getDefaultPath(file) {
+        Object.getOwnPropertyNames(params).forEach(function (p) {
+            args.push(p);
+
+            if (params[p]) {
```
What happens if the param value is false. i don't know if there will be ever a case like that for the signer though

rwmtse replied:
If params is null/undefined, this means no additional params will be used for blackberry-signer.

adrianlee commented:
Okay, everything looks good to me but there seems to be missing a test case for the nativepackager params.

rwmtse commented:
Right, let me add one for native packager

adrianlee commented:
r+ code review complete

rwmtse commented:
@issp Please use this build for your testing:
http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-params-tools/7/

jengee commented:
Feature testing completed. All testcases passed.

ishneur commented:
Tested on Windows and Mac.

There are a couple of issues that I noticed. Not sure whether they are considered bugs or not:
1. If in my .json file I specify params for blackberry-nativepackager and tell it to install the app, during packaging both the simulator bar and the device bar are deployed, with the simulator bar being deployed first and then the device bar overwriting it. Not sure whether there is a way to prevent it in the json file. This is the output:
   [INFO]    Populating application source
   [INFO]    Parsing config.xml
   [INFO]    Generating output files
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\simulator\
   APISmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Resetting stale session
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1150376
   [INFO]    Info: Installing ...Info: Processing 1150376 bytes
   [INFO]    Info: Progress 7%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7
   [INFO]    actual_id::testRel_PISmokeTest1db56dc7actual_version::1.2.3.16result::
   success
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\device\API
   SmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1137320
   [INFO]    Info: Installing ...Info: Processing 1137320 bytes
   [INFO]    Info: Progress 40%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7actual_id::testR
   el_PISmokeTest1db56dc7
   [INFO]    actual_version::1.2.3.16result::success
   [INFO]    Info: Bar signed.
   [INFO]    BAR packaging complete
2. If in my json file I call both blackberry-signer and blackberry-nativepackager to sign and deploy, it looks like the bar is deployed before signing, so the bar deployed to the device is unsigned. It can also be seen in the above output.

The final bar on my desktop contains "Package-Author-Certificate-Hash: mud1WyQkseGkPFj3LeLWKETe0oxPTZNoni1Aa86Im4Nc1rqo8uDwpRWlGI4A4T3rr9epSKYlqGc76hp3h94fcA" in the manifest file, but the bar file on the device doesn't have it in the manifest (I checked it via ssh since my device is not secure). The final bar also contains AUTHOR and RDK files, but I don't see those on the device inside the bar.

The json file I used is:
{
    "_comment": "You can specify additional parameters to be passed to downstream tools by using the bbwp --params option",
    "blackberry-signer": {
        "-proxyhost": "wps01-vmyyz.labyyz.testnet.rim.net",
        "-proxyport": "80",
        "-proxyusername": "user01",
        "-proxypassword": "password"
    },
    "blackberry-nativepackager": {
        "-installApp": "",
        "-device": "10.137.234.255",
        "-password": "qaqa"
    }
}

The signing test was more relevant on Mac than on Windows, since even if Windows is configured to use a proxy, signing still works without specifying parameters.

rwmtse commented:
@kjwallis @nukulb see Igor's test results. If we want this feature to work for more cases, we might need to take into consideration for more things, like allowing params to be specified for simulator/device build etc.

----- Original Message -----
From: Igor Shneur [mailto:reply@reply.github.com]
Sent: Saturday, July 07, 2012 06:00 PM
To: Rosa Tse
Subject: Re: [BB10-Webworks-Packager] Pull Req: Added --params option to pass additional params to downstream tools (#65)

Tested on Windows and Mac.

There are a couple of issues that I noticed. Not sure whether they are considered bugs or not:
1. If in my .json file I specify params for blackberry-nativepackager and tell it to install the app, during packaging both the simulator bar and the device bar are deployed, with the simulator bar being deployed first and then the device bar overwriting it. Not sure whether there is a way to prevent it in the json file. This is the output:
   [INFO]    Populating application source
   [INFO]    Parsing config.xml
   [INFO]    Generating output files
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\simulator\
   APISmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Resetting stale session
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1150376
   [INFO]    Info: Installing ...Info: Processing 1150376 bytes
   [INFO]    Info: Progress 7%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7
   [INFO]    actual_id::testRel_PISmokeTest1db56dc7actual_version::1.2.3.16result::
   success
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\device\API
   SmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1137320
   [INFO]    Info: Installing ...Info: Processing 1137320 bytes
   [INFO]    Info: Progress 40%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7actual_id::testR
   el_PISmokeTest1db56dc7
   [INFO]    actual_version::1.2.3.16result::success
   [INFO]    Info: Bar signed.
   [INFO]    BAR packaging complete
2. If in my json file I call both blackberry-signer and blackberry-nativepackager to sign and deploy, it looks like the bar is deployed before signing, so the bar deployed to the device is unsigned. It can also be seen in the above output.

The final bar on my desktop contains "Package-Author-Certificate-Hash: mud1WyQkseGkPFj3LeLWKETe0oxPTZNoni1Aa86Im4Nc1rqo8uDwpRWlGI4A4T3rr9epSKYlqGc76hp3h94fcA" in the manifest file, but the bar file on the device doesn't have it in the manifest (I checked it via ssh since my device is not secure). The final bar also contains AUTHOR and RDK files, but I don't see those on the device inside the bar.

The json file I used is:
{
    "_comment": "You can specify additional parameters to be passed to downstream tools by using the bbwp --params option",
    "blackberry-signer": {
        "-proxyhost": "wps01-vmyyz.labyyz.testnet.rim.net",
        "-proxyport": "80",
        "-proxyusername": "user01",
        "-proxypassword": "password"
    },
    "blackberry-nativepackager": {
        "-installApp": "",
        "-device": "10.137.234.255",
        "-password": "qaqa"
    }
}

The signing test was more relevant on Mac than on Windows, since even if Windows is configured to use a proxy, signing still works without specifying parameters.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/65#issuecomment-6826924

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

jengee commented:
Nice catch. I didn't think of using it in that way. I guess it is always good to have more than 1 person to look at it.

# -j

Sent from a Blackberry device while not driving.

----- Original Message -----
From: Igor Shneur [mailto:reply@reply.github.com]
Sent: Saturday, July 07, 2012 06:00 PM
To: Jenny Gee
Subject: Re: [BB10-Webworks-Packager] Pull Req: Added --params option to pass additional params to downstream tools (#65)

Tested on Windows and Mac.

There are a couple of issues that I noticed. Not sure whether they are considered bugs or not:
1. If in my .json file I specify params for blackberry-nativepackager and tell it to install the app, during packaging both the simulator bar and the device bar are deployed, with the simulator bar being deployed first and then the device bar overwriting it. Not sure whether there is a way to prevent it in the json file. This is the output:
   [INFO]    Populating application source
   [INFO]    Parsing config.xml
   [INFO]    Generating output files
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\simulator\
   APISmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Resetting stale session
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1150376
   [INFO]    Info: Installing ...Info: Processing 1150376 bytes
   [INFO]    Info: Progress 7%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7
   [INFO]    actual_id::testRel_PISmokeTest1db56dc7actual_version::1.2.3.16result::
   success
   [INFO]    Info: Package created: C:\WebWorks\Widgets\BB10\API_Testing\device\API
   SmokeTest.bar
   [INFO]    Info: Sending request: Install
   [INFO]    Info: Action: Install
   [INFO]    Info: File size: 1137320
   [INFO]    Info: Installing ...Info: Processing 1137320 bytes
   [INFO]    Info: Progress 40%...
   [INFO]    Info: Progress 100%...
   [INFO]    Info: Progress 100%...
   [INFO]    actual_dname::APISmokeTest.testRel_PISmokeTest1db56dc7actual_id::testR
   el_PISmokeTest1db56dc7
   [INFO]    actual_version::1.2.3.16result::success
   [INFO]    Info: Bar signed.
   [INFO]    BAR packaging complete
2. If in my json file I call both blackberry-signer and blackberry-nativepackager to sign and deploy, it looks like the bar is deployed before signing, so the bar deployed to the device is unsigned. It can also be seen in the above output.

The final bar on my desktop contains "Package-Author-Certificate-Hash: mud1WyQkseGkPFj3LeLWKETe0oxPTZNoni1Aa86Im4Nc1rqo8uDwpRWlGI4A4T3rr9epSKYlqGc76hp3h94fcA" in the manifest file, but the bar file on the device doesn't have it in the manifest (I checked it via ssh since my device is not secure). The final bar also contains AUTHOR and RDK files, but I don't see those on the device inside the bar.

The json file I used is:
{
    "_comment": "You can specify additional parameters to be passed to downstream tools by using the bbwp --params option",
    "blackberry-signer": {
        "-proxyhost": "wps01-vmyyz.labyyz.testnet.rim.net",
        "-proxyport": "80",
        "-proxyusername": "user01",
        "-proxypassword": "password"
    },
    "blackberry-nativepackager": {
        "-installApp": "",
        "-device": "10.137.234.255",
        "-password": "qaqa"
    }
}

The signing test was more relevant on Mac than on Windows, since even if Windows is configured to use a proxy, signing still works without specifying parameters.

---

Reply to this email directly or view it on GitHub:
https://github.com/blackberry-webworks/BB10-Webworks-Packager/pull/65#issuecomment-6826924

---

This transmission (including any attachments) may contain confidential information, privileged material (including material protected by the solicitor-client or other applicable privileges), or constitute non-public information. Any use of this information by anyone other than the intended recipient is prohibited. If you have received this transmission in error, please immediately reply to the sender and delete this information from your system. Use, dissemination, distribution, or reproduction of this transmission by unintended recipients is not authorized and may be unlawful.

kwallis commented:
We can leave this feature as is.  The goal was a simple pass through.  If we want to handle the actual use case of installing the app with bbwp it should be done in a different way.

nukulb commented:
@rwmtse - rebase please

nukulb commented:
@ishneur / @issp - is this tested fully now?
Ready for merge?

ishneur commented:
Retested this on Windows using build http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-next-params-tools/8/ and no new issues found.

I already tested it before the rebase and posted my comments above. The same issues still apply.

ishneur commented:
If I specify a params file which doesn't exist, then packaging fails but the packager still creates an "src" and a "simulator" folder in the directory in which the zip file is and doesn't clean them up.

This is a packager problem not directly related to this feature, so we can just open a separate issue for this. It could be related to blackberry/BB10-Webworks-Packager#51.

tohman commented:
@rwmtse - can you please rebase into one commit so that we can merge easily? right now we are getting merge conflicts

rwmtse committed:
Implements --params option for bbwp

Fixes issue blackberry/BB10-Webworks-Packager#157

Reviewed By: Adrian Z Lee <adrilee@rim.com>
Tested By: Igor Shneur <ishneur@rim.com>

rwmtse commented:
@tohman @nukulb Rebased, squashed, smoked tested

nukulb commented:
merged manually
